src/features/registry/registry.ts

```diff
@@ -297,37 +422,79 @@ const authorizeRequest = async (
 					name,
 					actions: _.intersection(requestedActions, allowedActions),
 				};
-			} else {
-				const match = name.match(NEW_REGISTRY_REGEX);
-				if (match != null) {
-					// request for new-style, authenticated v2/randomhash image
-					let effectiveName = name;
-					if (match[4] != null) {
-						// This is a multistage image, use the root image name
-						effectiveName = match[3];
-					}
-					return await resolveAccess(
-						req,
-						type,
-						name,
-						effectiveName,
-						requestedActions,
-						undefined,
-						tx,
-					);
-				} else {
-					// request for legacy public-read appName/commit image
+			}
+
+			let match = name.match(NEW_REGISTRY_REGEX);
+			if (match != null) {
+				// request for new-style, authenticated v2/randomhash image
+				let effectiveName = name;
+				if (match[4] != null) {
+					// This is a multistage image, use the root image name
+					effectiveName = match[3];
+				}
+				return await resolveAccess(
+					req,
+					type,
+					name,
+					effectiveName,
+					requestedActions,
+					undefined,
+					undefined,
+					tx,
+				);
+			}
+
+			// match <org>/<app>/<semver|commit>/<service> where <semver|commit> and <service> are optional
+			match = name.match(APP_RELEASE_REGEX);
+			if (match != null && match.length > 3) {
+				// request for application (blocks) releases
+				let alias;
+
+				const application = match[1];
+				let release = match[4] || undefined;
+				const service = match[5];
+
+				// allow keywords like 'latest' and 'current' to return the target release for the application
+				if (release == null || TARGET_RELEASE_KEYWORDS.includes(release)) {
+					release = undefined;
+				}
+
+				const imageLocation = await resolveImageLocation(
+					application,
+					release,
+					service,
+					tx,
+				);
+
+				if (imageLocation != null) {
```

Comment:
Shouldn't it be like: 
```suggestion
				if (imageLocation == null) {
					return {
						type,
						name,
						actions: [],
					};
				}
```
Does it make to proceed with the `// request for legacy public-read appName/commit image` default case?

Reply:
sounds good!

Reply:
From the tests that old format only applies to registry v1, which we don't have in balenaCloud afaik but maybe still exists in OB?

Reply:
Good catch, the registry v1 tests aren't passing because
`test_acl1/d6cd1e2bd19e03a81132a23b2025920577f84e37`
matches the same regex as
`<org>/<app>/<?version>`
so I need to figure out if that legacy path has any valid use case these days

Reply:
Keeping this change as additive for now, so if we can't resolve an image path we will fall back to the legacy `appname/commit` logic (the regex matches both unfortunately).